addons/web/static/src/legacy/js/widgets/model_field_selector_popover.js

```diff
@@ -0,0 +1,732 @@
+                    } else if (full_height < el_position + el_height) {
+                        $page.scrollTop(current_scroll + el_height);
+                    }
+                }
+                break;
+            case $.ui.keyCode.RIGHT:
+                if (inputHasFocus) break;
+                e.preventDefault();
+                if (name) {
+                    var field = this._getLastPageField(name);
+                    if (field.relation) {
+                        this._goToNextPage(field);
+                    } else if (this.options.needDefaultValue && !this._isOnDefaultValuePage()) {
+                        this._goToDefaultValuePage(name);
+                    }
+                }
+                break;
+            case $.ui.keyCode.LEFT: {
+                if (inputHasFocus) break;
+                e.preventDefault();
+                const isOnDefaultValuePage = this._isOnDefaultValuePage();
+                this._goToPrevPage();
+                if (isOnDefaultValuePage) {
+                    this.$searchInput.focus();
+                }
+                break;
+            }
+            case $.ui.keyCode.ESCAPE:
+                e.stopPropagation();
+                this._hidePopover(this.options.cancelOnEscape);
+                break;
+            case $.ui.keyCode.ENTER:
+                e.preventDefault();
+                if (!name && !this._isOnDefaultValuePage()) {
+                    this._hidePopover(this.options.cancelOnEscape);
+                    break;
+                }
+                if (this.options.needDefaultValue && this._isOnDefaultValuePage()) {
+                    this._hidePopover();
+                } else if (this.options.needDefaultValue) {
+                    this._goToDefaultValuePage(name);
+                } else {
+                    this._selectField(this._getLastPageField(name));
+                }
+                break;
+        }
+    }
+});
+
+return ModelFieldSelectorPopOver;
+
+/**
+ * Allows to transform a mapping field name -> field info in an array of the
+ * field infos, sorted by field user name ("string" value). The field infos in
+ * the final array contain an additional key "name" with the field name.
+ *
+ * @param {Object} fields - the mapping field name -> field info
+ * @param {string} model
+ * @returns {Object[]} the field infos sorted by field "string" (field infos
+ *                     contain additional keys "model" and "name" with the field
+ *                     name)
+ */
+function sortFields(fields, model, order) {
+    var array = _.chain(fields)
+        .pairs()
+        .sortBy(function (p) { return p[1].string; });
+    if (order !== 'string') {
+        array = array.sortBy(function (p) {return p[1][order]; });
+    }
+    return array.map(function (p) {
+            return _.extend({
+                name: p[0],
+                model: model,
+            }, p[1]);
+        }).value();
+}
+});
```

Comment:
That would have been great to have a separate commit with only the move of these lines, even if the resulting code was broken. That way I could have seen what were the changes that were needed to make it work in the new system. Here I cannot see anything unless I review the entire file, including the old code... 😢